### ISSUE TITLE
KDLoss and KLDiscretLoss difference , KDLoss mistake 

### ISSUE THREAD

**Issue by CatDroid** [IDEA-Research/DWPose#92]
in KLDiscretLoss (classification_loss.py)
log_pt's or labels's shape is [B*K, width] or [B*K, height]
width or height is ‘number of categories'
```
def criterion(self, dec_outs, labels):
   ......
  loss = torch.mean(self.kl_loss(log_pt, labels), dim=1)  
   ......
```
it will calculate the mean KL divergence of 'a key point x coordinate'  on ' all categories'(along the dim=1 dimension)

but in KDLoss  (kd.py)
```
def loss(self, logit_s, logit_t, beta, weight):
    ......
    loss_all = torch.sum(self.kl_loss(s_i, t_i), dim=1)
    ......
```
it will calculate  the sum of KL divergence along  dim=1 dimension .

does doing this have a different meaning?

Beside, in 'def loss(self, logit_s, logit_t, beta, weight):'  if 'if len(logit_s.shape) == 3:' not true, K is not defined , maybe a mistake ?
